src/timedate/timedated.c

```diff
@@ -938,6 +942,11 @@ static int method_set_ntp(sd_bus_message *m, void *userdata, sd_bus_error *error
         /* This method may be called frequently. Forget the previous job if it has not completed yet. */
         LIST_FOREACH(units, u, c->units)
                 u->path = mfree(u->path);
+        /* Pretend success to forgotten requestors, only the last will wait properly */
+        if (c->ntp_req) {
+                (void) sd_bus_reply_method_return(c->ntp_req, NULL);
+                c->ntp_req = sd_bus_message_unref(c->ntp_req);
+        }
```

Comment:
this doesn't feel right: leaving the client hanging like this.

just turn `ntp_req` into a Set instead, and then reply to them all once we get JobRemoved?

bus_message_hash_ops already exists, it's trivial to just add the correctness here and keep track of all messages.

consider returning an early error if the set grows too large however, and refuse accepting new requests.

Reply:
> if the set grows too large

Can the set size <= 1? :-p

Reply:
i mean, the comment says it might be called "frequently". i wonder in which context this would ever be called "frequently"... but if the comment is right, the size if 1 is probably a bit too low ;-)